meilisearch/src/search.rs

```diff
@@ -523,6 +553,20 @@ fn insert_geo_distance(sorts: &[String], document: &mut Document) {
     }
 }
 
+fn insert_semantic_score(query: &[f32], vectors: Value, document: &mut Document) {
+    let vectors =
+        match serde_json::from_value(vectors).map(VectorOrArrayOfVectors::into_array_of_vectors) {
+            Ok(vectors) => vectors,
+            Err(_) => return,
+        };
+    let similarity = vectors
+        .into_iter()
+        .map(|v| OrderedFloat(dot_product_similarity(query, &v)))
+        .max()
+        .map(OrderedFloat::into_inner);
+    document.insert("_semanticScore".to_string(), json!(similarity));
```

Comment:
this way of inserting the field will replace any original value from the document for the field `_semanticScore`.

This behavior is consistent with `_geoDistance`, but inconsistent with `_rankingScore`, `_rankingScoreDetails`, `_formatted` and `_matchesPosition`, that just output both fields.

To be consistent with the latter fields, you'd need to edit the `SearchHit` structure to look like:

```rust
#[derive(Debug, Clone, Serialize, PartialEq)]
pub struct SearchHit {
    #[serde(flatten)]
    pub document: Document,
    #[serde(rename = "_formatted", skip_serializing_if = "Document::is_empty")]
    pub formatted: Document,
    #[serde(rename = "_matchesPosition", skip_serializing_if = "Option::is_none")]
    pub matches_position: Option<MatchesPosition>,
    #[serde(rename = "_rankingScore", skip_serializing_if = "Option::is_none")]
    pub ranking_score: Option<f64>,
    #[serde(rename = "_rankingScoreDetails", skip_serializing_if = "Option::is_none")]
    pub ranking_score_details: Option<serde_json::Map<String, serde_json::Value>>,
    #[serde(rename = "_semanticScore", skip_serializing_if = "Option::is_none")]
    pub semantic_score: Option<f32>,
}
```

And then pass the value in `Some` for the `SearchHit` corresponding to each document with a similarity score.